src/register/pmpcfgx.rs

```diff
@@ -1,7 +1,129 @@
 /// Physical memory protection configuration
+
 pub mod pmpcfg0 {
-    read_csr_as_usize!(0x3A0, __read_pmpcfg0);
-    write_csr_as_usize!(0x3A0, __write_pmpcfg0);
+    use bit_field::BitField;
+
+    #[derive(Clone, Copy, Debug)]
+    pub enum Permission {
+        NONE = 0,
+        R = 1,
+        W = 2,
+        RW = 3,
+        X = 4,
+        RX = 5,
+        WX = 6,
+        RWX = 7,
+    }
```

Comment:
It seems that having `X` only or `WX` may not make sense.  If you can write and execute from the address, why couldn't you read it too?  However, it seems to be possible to use those values, so I'm not sure.

Reply:
Let's keep them for completeness.

Reply:
sure